joomla/joomla-cms
# [Testing] Add basic setup and sample tests for Codeception System Tests

### Summary of Changes

This pull request adds the basic setup and configuration for the Joomla browser based System Tests. It contains sample tests for com_users. During the next weeks more tests are going to be ported from the [system tests repository
](https://github.com/joomla-projects/gsoc16_browser-automated-tests). Additionally Continuous Integration will be activated running the test suite for every PR and commit.

### Testing Instructions

For a complete description check the README.md in tests/codeception

Prerequisites:

* Webserver with PHP installed
* Google Chrome
* Composer
* PHP 5.6 Cli (or later)
* Java JDK (8 or later)

Step by Step:

* Navigate in your terminal to `tests/codeception`
* Do a `composer install`
* cp and modify the `acceptance.suite.dist.yml` to `acceptance.suite.yml`
* Navigate back to the root folder of Joomla
* Run `tests/codeception/vendor/bin/robo run:tests`

### Expected result

Joomla has System tests

### Actual result

Joomla has no System tests

### Documentation Changes Required

Yes, parts are already integrated and will be updated the next days.

Thank you to all people working on this topic in the last years!
